debugger/src/debugger.rs

```diff
@@ -537,16 +537,25 @@ impl Debugger {
                                 });
                                 debug_adapter.send_event("stopped", event_body);
                             }
-                            //TODO: Need to implement LockedUp in probe-rs/src/debug
-                            // CoreStatus::LockedUp => {
-                            //     debug_adapter.send_response::<()>(
-                            //         &request,
-                            //         Err(DebuggerError::Other(anyhow!(
-                            //             "The processor is in LOCKED status, as a result of an unrecoverable error"
-                            //         ))),
-                            //     );
-                            //     return false;
-                            // }
+                            CoreStatus::LockedUp => {
+                                let event_body = Some(StoppedEventBody {
+                                    reason: new_status.short_long_status().0.to_owned(),
+                                    description: Some(new_status.short_long_status().1.to_owned()),
+                                    thread_id: Some(core_data.target_core.id() as i64),
+                                    preserve_focus_hint: Some(false),
+                                    text: None,
+                                    all_threads_stopped: Some(true),
+                                    hit_breakpoint_ids: None,
+                                });
+                                debug_adapter.send_event("stopped", event_body);
+                                debug_adapter.send_response::<()>(
+                                    &request,
+                                    Err(DebuggerError::Other(anyhow!(
```

Comment:
I don't think this should be considered an error. The debugger / probe-rs is running normally, it is just the core who is in a certain state.

Also, can we send a response without a request?

Reply:
- The way the DAP debugger is implemented currently is that it lives for a single session on the core. If anything unrecoverable happens on the core, the debugger session is considered complete. When the user starts a new debug session (maybe after a chip reset), then a new instance of the debugger is created. 

- I consider it an error, because the code that is being debugged (user code on the embedded device) resulted in an exception. I try to be explicit with the error message, so that the user can tell who's to blame.  There are actually very few times when these types of (caused by the user code) shows up, so I'm open to suggestions on how you'd like to handle it. At the end of the day, I think it makes sense for the debugger to interrupt the debugging session in this case, since there is no sane path forward.

- At this point in the code, there is always a request in play, even if it is the result of ongoing polling for core status, rather than a user initiated request.  VSCode handles these just fine :)

Reply:
Does VSCode send a request to poll for the core status? I thought that it just waits for the event. The `CoreStatus::Halted` case does not send a response, so this seems a bit weird. Just because VSCode accepts it doesn't mean it's right :wink:

Reply:
>     * I consider it an error, because the code that is being debugged (user code on the embedded device) resulted in an exception. I try to be explicit with the error message, so that the user can tell who's to blame.  There are actually very few times when these types of (caused by the user code) shows up, so I'm open to suggestions on how you'd like to handle it. At the end of the day, I think it makes sense for the debugger to interrupt the debugging session in this case, since there is no sane path forward.

From what I understand, the event already should stop the debugger session, and indicate what happened. What's the benefit of the additional error response?

Reply:
Absolutely correct ... thanks for catching that!